### DISCUSSION TITLE
[Feature Request] Append query string to URLs of files in local volumes to bust the browser's cache when the file changes

### DISCUSSION THREAD

**Discussion by benface** [craftcms/cms#8802] [Ideas]
It seems the Amazon S3 plugin (and maybe others) already do this, but not the core Local volume type. When someone replaces an image in a local volume, and the front-end of the site displays that image like this: `<img src="{{ image.url }}">`, the URL remains the same, meaning that if the image was already cached by a browser, it will keep displaying the old image until the cache is cleared. It would be nice if `getUrl()` appended a query string representing the file's last modified date or the file's hash to force the browser to re-download the image.

**Comment by juddlyon** [comment]
Agreed - I have authors that run into this and I have to tell them to rename the file.

**Comment by brandonkelly** [comment]
We can consider adding this to core, but in the meantime you can use the [Asset Rev](https://plugins.craftcms.com/assetrev) plugin to accomplish this.

Or just manually add a query string param to the URLs using the [url()](https://docs.craftcms.com/v3/dev/functions.html#url) function:

```twig
<img src="{{ url(image.url, {v: image.dateModified.timestamp}) }}">

**Comment by benface** [comment]
Thanks @brandonkelly, unfortunately the Asset Rev plugin doesn't really support that use case at the moment but I created a FR there too: https://github.com/clubstudioltd/craft-asset-rev/issues/27

**Comment by brandonkelly** [comment]
@benface just edited my last comment, fyi

**Comment by benface** [comment]
Oh that's a nice solution, I would suggest adding this to the docs @mattstein.
